tests/functional/openstackdataplanedeployment_controller_test.go

```diff
@@ -30,23 +38,116 @@ var _ = Describe("Dataplane Deployment Test", func() {
 			Namespace: namespace,
 			Name:      "dataplane-ansible-ssh-private-key-secret",
 		}
+		neutronOvnMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "neutron-ovn-metadata-agent-neutron-config",
+		}
+		novaNeutronMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-metadata-neutron-config",
+		}
+		novaCellComputeConfigSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-cell1-compute-config",
+		}
 		err := os.Setenv("OPERATOR_SERVICES", "../../config/services")
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	When("A dataplaneDeployment is created with matching NodeSet", func() {
 		BeforeEach(func() {
-			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNoNodeSetSpec()))
-			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 			CreateSSHSecret(dataplaneSSHSecretName)
+			CreateSSHSecret(neutronOvnMetadataSecretName)
+			CreateSSHSecret(novaNeutronMetadataSecretName)
+			CreateSSHSecret(novaCellComputeConfigSecretName)
```

Comment:
We should rename this function if we want to use it to create generic secrets. Although, do we already have a helper function for this in lib-common? Maybe we can just re-use something there and remove our bespoke implementation of this logic.

Reply:
These are not SSH secrets so they don't have to have `ssh-privatekey` field, so I think simply reusing th.CreateSecret is enough as the dataplane operator really don't care about the content of these secrets. I even doubt that it care about the existence of these secrets.

Reply:
I was wrong about the latter. The existence of the secret is important that dataplane looks into the Secret to to the volume mount for AnsibleEE. (Probably can be removed from the impl but that is a separate issue from this PR)